contracts/NFTStakingManager.sol

```diff
@@ -522,16 +668,20 @@ contract NFTStakingManager is
     emit PrepaidCreditsAdded(hardwareOperator, licenseHolder, creditSeconds);
   }
 
-  function processProof(uint32 messageIndex) public {
+  function processProof(bytes32 validationID, uint32 messageIndex) public {
     bytes32 uptimeBlockchainID = 0x0000000000000000000000000000000000000000000000000000000000000000;
-    (bytes32 validationID, uint64 uptimeSeconds) = ValidatorMessages.unpackValidationUptimeMessage(
-      _getPChainWarpMessage(messageIndex, uptimeBlockchainID).payload
-    );
+
+    (bytes32 uptimeValidationID, uint64 uptimeSeconds) = ValidatorMessages
+      .unpackValidationUptimeMessage(_getPChainWarpMessage(messageIndex, uptimeBlockchainID).payload);
+
+    if (uptimeValidationID != validationID) {
```

Comment:
Why this check? The proof will have the validationID in it already, what do we gain by having the caller also pass it in separately?